compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -127,6 +128,22 @@ public STNode parseAsStatement() {
         return stmt;
     }
 
+    /**
+     * Completely parses a given input as a block statement.
+     *
+     * @return Parsed node
+     */
+    public STNode parseAsBlockStatement() {
+        startContext(ParserRuleContext.COMP_UNIT);
+        startContext(ParserRuleContext.FUNC_DEF);
+        startContext(ParserRuleContext.FUNC_BODY_BLOCK);
+        startContext(ParserRuleContext.WHILE_BLOCK);
```

Comment:
Any reason for starting `ParserRuleContext.WHILE_BLOCK`?

Reply:
if we check all places where `parseBlockNode()` is called. There's a parent context. `WHILE_BLOCK`, `IF_BLOCK`, `FOREACH_STMT`, `LOCK_STMT` etc. We can keep `FUNC_BODY_BLOCK` only since inside parseFunctionBodyBlock() we don't call `parseBlockNode() ` directly right